wazo_confgend/generators/extensionsconf.py

```diff
@@ -51,49 +55,32 @@
 }
 
 
-class CustomConfigParserStorage(DictMixin):
-    def __init__(self, init=None):
-        self._data = init if init else []
-
-    def __setitem__(self, key, value):
-        # This reverse the config parser logic of merging option keys
-        if isinstance(value, list):
-            if len(value) != 1:
-                raise RuntimeError("Unexpected value from configparser")
-            value = value[0]
-        self._data.append((key, value))
-
-    def __getitem__(self, key):
-        values = [v for k, v in self._data if k == key]
-        if len(values) > 1:
-            raise RuntimeError("Can't get a key with multiple value")
-        if values:
-            return values[0]
-        else:
-            raise KeyError
-
-    def __delitem__(self, key):
-        n_items = len(self._data)
-        self._data = [(k, v) for k, v in self._data if k != key]
-        if len(self._data) == n_items:
-            raise KeyError
-
-    def keys(self):
-        return [k for k, v in self._data]
+class CustomConfigParserStorage(collections.UserDict):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.data = collections.OrderedDict(self.data)
+        self.counter = itertools.count(start=1)
 
-    def values(self):
-        return [v for k, v in self._data]
-
-    def __iter__(self):
-        for k, v in self._data:
-            yield k
+    def __setitem__(self, k, v):
+        if isinstance(v, list):
+            # if list, we assume this is a new option declaration
+            # configparser creates option values as list to deal with multiline values
+            self.data[f"{next(self.counter)}:{k}"] = v
```

Comment:
Is a simple `:` a safe enough separator? Perhaps it would safer to prefix the key with something instead of just that.
Otherwise. 
Also, a perhaps simpler/safer option. Couldn't we simply store multiple values as another type? The config parser only re-assembles values in lists, specifically (it does `isinstance(val, list)`). So we could create a class for our use to be able to easily identify it or simply used a tuple or something?